### PR TITLE
Added the BankTransaction Savina benchmark

### PR DESCRIPTION
**Pull request by EliasC**
This PR subsumes #761.

### CODE REVIEW

**Comment by supercooldave** [comment]
It would definitely be interesting to see if the run-time will improve with the new future implementation. And/or whether this is a better way of doing this -- there are a lot of gets at the end of methods where the value of returned by the future is not used. Perhaps there's a better way.

**Comment by supercooldave** [comment]
(My comment shouldn't affect this PR.)

**Comment by EliasC** [comment]
> there are a lot of gets at the end of methods where the value of returned by the future is not used

The way this is done in the original benchmark is through selective message receives to make sure that no other transactions are started before the current one finishes. Benchmarking against new futures or the implementation of the `atomic` block would indeed be interesting! (ping @duckyduckmaster @M34ris).

File: modules/standard/Random.enc [hunk 1]
```diff
@@ -37,4 +37,9 @@ read class Random
 
     END
   end
```
Comment (supercooldave) [inline modules/standard/Random.enc:41]: Does your return key not work?

**Review by supercooldave — COMMENTED** [review]
One terribly minor comment, but there should be newlines between function definitions.

**Comment by supercooldave** [comment]
One minor comment. I'm happy to accept this into the fold after it is addressed.

**Comment by supercooldave** [comment]
Merging at 16:00 unless there are objections.

File: modules/standard/Random.enc (hunk 1 above)
Comment (EliasC) [reply modules/standard/Random.enc:41]: This code was translated automatically at some point, and the pretty-printer adds an extra newline before an `END` (for some reason)

Comment (EliasC) [reply modules/standard/Random.enc:41]: (and does not add newlines between methods)

Comment (supercooldave) [reply modules/standard/Random.enc:41]: I fixed it all up by hand. 
(Didn't the return key work when writing the pretty printer? :smiley:)

Comment (EliasC) [reply modules/standard/Random.enc:41]: Thanks! 😃

**Pull request merged**